test/e2e/garbagecollector/garbagecollector_test.go

```diff
@@ -0,0 +1,478 @@
+	"sigs.k8s.io/yaml"
+
+	configcrds "github.com/kcp-dev/kcp/config/crds"
+	"github.com/kcp-dev/kcp/config/helpers"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/test/e2e/fixtures/apifixtures"
+	wildwestv1alpha1 "github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest/apis/wildwest/v1alpha1"
+	wildwestclientset "github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestGarbageCollectorBuiltInCoreV1Types(t *testing.T) {
+	t.Parallel()
+
+	server := framework.SharedKcpServer(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	cfg := server.BaseConfig(t)
+
+	kubeClusterClient, err := kubernetes.NewForConfig(cfg)
+	require.NoError(t, err, "error creating kube cluster client")
+
+	orgClusterName := framework.NewOrganizationFixture(t, server)
+
+	ws := framework.NewWorkspaceFixture(t, server, orgClusterName, framework.WithName("gc-%d", 1))
+
+	t.Logf("Creating owner configmap")
+	owner, err := kubeClusterClient.CoreV1().ConfigMaps("default").
+		Apply(logicalcluster.WithCluster(ctx, ws),
+			corev1ac.ConfigMap("owner", "default"),
+			metav1.ApplyOptions{FieldManager: "e2e-test-runner"})
+	require.NoError(t, err, "Error applying owner configmap %s|default/owner: %v", ws, err)
+
+	t.Logf("Creating owned configmap")
+	owned, err := kubeClusterClient.CoreV1().ConfigMaps("default").
+		Apply(logicalcluster.WithCluster(ctx, ws),
+			corev1ac.ConfigMap("owned", "default").
+				WithOwnerReferences(metav1ac.OwnerReference().
+					WithAPIVersion("v1").
+					WithKind("ConfigMap").
+					WithName(owner.Name).
+					WithUID(owner.UID)),
+			metav1.ApplyOptions{FieldManager: "e2e-test-runner"})
+	require.NoError(t, err, "Error applying owned configmap %s|default/owned: %v", ws, err)
+
+	t.Logf("Deleting owner configmap")
+	err = kubeClusterClient.CoreV1().ConfigMaps("default").
+		Delete(logicalcluster.WithCluster(ctx, ws), owner.Name, metav1.DeleteOptions{})
+
+	t.Logf("Waiting for the owned configmap to be garbage collected")
+	framework.Eventually(t, func() (bool, string) {
+		_, err = kubeClusterClient.CoreV1().ConfigMaps("default").
+			Get(logicalcluster.WithCluster(ctx, ws), owned.Name, metav1.GetOptions{})
+		return apierrors.IsNotFound(err), fmt.Sprintf("configmap not garbage collected: %s", owned.Name)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "error waiting for owned configmap to be garbage collected")
+}
+
+func TestGarbageCollectorTypesFromBinding(t *testing.T) {
+	t.Parallel()
+
+	server := framework.SharedKcpServer(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	orgClusterName := framework.NewOrganizationFixture(t, server)
+
+	apiProviderClusterName := framework.NewWorkspaceFixture(t, server, orgClusterName)
```

Comment:
Global comment: add `framework.WithName(...)` as the last param to name each child workspace when calling `framework.NewWorkspaceFixture`

Reply:
Done with 76d0270e4a684995d52a07567734f82aa72b9af7.